plugins/modules/win_domain_ou.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_domain_ou
+short_description: Manage Active Directory Organizational Units
+author: ['Joe Zollo (@joezollo)', 'Larry Lane (@gamethis)']
+version_added: 1.8.0
+requirements:
+  - This module requires Windows Server 2012 or Newer
+description:
+  - Manage Active Directory Organizational Units
+  - Adds, Removes and Modifies Active Directory Organizational Units
+  - Task should be delegated to a Windows Active Directory Domain Controller
+options:
+  name:
+    description:
+      - The name of the Organizational Unit
+    type: str
+    required: true
+  protected:
+    description:
+      - Indicates whether to prevent the object from being deleted. When this
+        l(protected=true), you cannot delete the corresponding object without
+        changing the value of the property.
+    type: bool
+    default: false
+  path:
+    description:
+      - Specifies the X.500 path of the OU or container where the new object is
+        created.
+      - defaults to adding ou at base of domain connected to.
+    type: str
+    required: false
+  state:
+    description:
+      - Specifies the desired state of the OU.
+      - When l(state=present) the module will attempt to create the specified
+        OU if it does not already exist.
+      - When l(state=absent), the module will remove the specified OU.
+      - When l(state=absent) and l(recursive=true), the module will remove all
+        the OU and all child OU's.
+    type: str
+    default: present
+    choices: [ present, absent ]
+  recursive:
+    description:
+      - Removes the OU and any child items it contains.
+      - You must specify this parameter to remove an OU that is not empty.
+    type: bool
+    default: false
+  domain_server:
+    description:
+    - Specifies the Active Directory Domain Services instance to connect to.
+    - Can be in the form of an FQDN or NetBIOS name.
+    - If not specified then the value is based on the domain of the computer
+      running PowerShell.
+    type: str
+  domain_username:
+    description:
+    - The username to use when interacting with AD.
+    - If this is not set then the user Ansible used to log in with will be
+      used instead when using CredSSP or Kerberos with credential delegation.
+    type: str
+  domain_password:
+    type: str
+    description:
+    - The password for the domain you are accessing
+  filter:
+    type: str
+    description: filter for lookup of ou.
+    default: '*'
+  properties:
+    type: dict
+    description:
+      - Free form dict of properties for the organizational unit. Follows LDAP property names, like StreetAddress or PostalCode.
```

Comment:
It's always nice to format some important entries with `C()`. Also a minor nit is that the properties are in `camelCase` not `PascalCase`.

```suggestion
      - Free form dict of properties for the organizational unit. Follows LDAP property names, like C(streetAddress) or C(postalCode).
```

Reply:
So vast majority of items come back in `PascalCase` ... a few come back `camelCase`

Sample output:
```shell
CanonicalName                   : ansible.test/VMware/End User Computing
City                            :
CN                              :
Country                         :
Created                         : 10/1/2021 5:05:44 PM
createTimeStamp                 : 10/1/2021 5:05:44 PM
Deleted                         :
Description                     :
DisplayName                     :
DistinguishedName               : OU=End User Computing,OU=VMware,DC=ansible,DC=test
dSCorePropagationData           : {1/1/1601 12:00:00 AM}
instanceType                    : 4
isDeleted                       :
LastKnownParent                 :
LinkedGroupPolicyObjects        : {}
ManagedBy                       :
Modified                        : 10/1/2021 5:05:44 PM
modifyTimeStamp                 : 10/1/2021 5:05:44 PM
Name                            : End User Computing
nTSecurityDescriptor            : System.DirectoryServices.ActiveDirectorySecurity
ObjectCategory                  : CN=Organizational-Unit,CN=Schema,CN=Configuration,DC=ansible,DC=test
ObjectClass                     : organizationalUnit
ObjectGUID                      : f6b8a10c-e5c5-4be8-9a89-1233cb1cad8f
ou                              : {End User Computing}
PostalCode                      :
ProtectedFromAccidentalDeletion : False
sDRightsEffective               : 15
State                           :
StreetAddress                   :
uSNChanged                      : 16501
uSNCreated                      : 16501
whenChanged                     : 10/1/2021 5:05:44 PM
whenCreated                     : 10/1/2021 5:05:44 PM
```